datasync/fetcher/fetcher.go

```diff
@@ -342,7 +342,9 @@ func (f *Fetcher) loop() {
 				if f.queueChangeHook != nil {
 					f.queueChangeHook(hash, true)
 				}
-				break insertFailed
+				// insertTasks channel is full, sleep a bit
+				time.Sleep(insertTasksWaitTime)
```

Comment:
How about setting a timer instead of `default`? Since the task has the exact block number the fetcher needs, it would be better to wait for a while instead of sending the task to the end of the queue. 
```
select {
case f.insertTasks <- insertTask{peer, block}: // do somthing
case <-time.NewTimer(insertTasksWaitTime).C:  // do the same things in `default`
}
```

Reply:
Don't mind too much because it's a small comment.

Reply:
Looks better. Thanks!